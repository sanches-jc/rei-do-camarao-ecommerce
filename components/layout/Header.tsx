import React, { type ReactElement } from "react";

export default function Header(): ReactElement {
  return (
    <header className="bg-white shadow-sm border-b border-gray-100">
      <div className="container mx-auto px-4 py-4">
        <div className="flex items-center justify-between">
          {/* Logo */}
          <a href="/" className="flex items-center space-x-2">
            <div className="text-2xl font-bold text-blue-600">
              Rei do Camarão
            </div>
          </a>

          {/* Navegação Principal */}
          <nav className="hidden md:flex space-x-8">
            <a
              href="/"
              className="text-gray-700 hover:text-blue-600 transition-colors duration-200 font-medium"
            >
              Home
            </a>
            <a
              href="/produtos"
              className="text-gray-700 hover:text-blue-600 transition-colors duration-200 font-medium"
            >
              Produtos
            </a>
            <a
              href="/sobre"
              className="text-gray-700 hover:text-blue-600 transition-colors duration-200 font-medium"
            >
              Sobre Nós
            </a>
            <a
              href="/contato"
              className="text-gray-700 hover:text-blue-600 transition-colors duration-200 font-medium"
            >
              Contato
            </a>
          </nav>

          {/* Ícones do Lado Direito */}
          <div className="flex items-center space-x-4">
            {/* Ícone de Carrinho */}
            <button
              type="button"
              aria-label="Abrir carrinho"
              className="p-2 text-gray-700 hover:text-blue-600 transition-colors"
            >
              <svg
                className="w-5 h-5"
                fill="none"
                stroke="currentColor"
                viewBox="0 0 24 24"
                aria-hidden="true"
              >
                <path
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth={2}
                  d="M3 3h2l.4 2M7 13h10l4-8H5.4M7 13L5.4 5M7 13l-2.293 2.293c-.63.63-.184 1.707.707 1.707H17m0 0a2 2 0 100 4 2 2 0 000-4zm-8 2a2 2 0 11-4 0 2 2 0 014 0z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </header>
  );
}
