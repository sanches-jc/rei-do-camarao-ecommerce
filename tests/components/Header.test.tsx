import React from "react";
import { render, screen } from "@testing-library/react";
import Header from "@/components/layout/Header";

describe("Header component", () => {
  test("renderiza o título do site", () => {
    render(<Header />);
    expect(screen.getByText(/Rei do Camarão/i)).toBeInTheDocument();
  });

  test("contém os links de navegação principais", () => {
    render(<Header />);
    expect(screen.getByRole("link", { name: /Home/i })).toBeInTheDocument();
    expect(screen.getByRole("link", { name: /Produtos/i })).toBeInTheDocument();
    expect(
      screen.getByRole("link", { name: /Sobre Nós/i }),
    ).toBeInTheDocument();
    expect(screen.getByRole("link", { name: /Contato/i })).toBeInTheDocument();
  });

  test("botão do carrinho existe e tem aria-label e ícone com aria-hidden", () => {
    render(<Header />);
    const cartButton = screen.getByRole("button", { name: /Abrir carrinho/i });
    expect(cartButton).toBeInTheDocument();

    const svg = cartButton.querySelector("svg");
    expect(svg).not.toBeNull();
    if (svg) {
      expect(svg).toHaveAttribute("aria-hidden", "true");
    }
  });

  test("links possuem hrefs corretos e nav tem classe para desktop", () => {
    render(<Header />);
    const home = screen.getByRole("link", { name: /Home/i });
    const produtos = screen.getByRole("link", { name: /Produtos/i });
    const sobre = screen.getByRole("link", { name: /Sobre Nós/i });
    const contato = screen.getByRole("link", { name: /Contato/i });

    expect(home).toHaveAttribute("href", "/");
    expect(produtos).toHaveAttribute("href", "/produtos");
    expect(sobre).toHaveAttribute("href", "/sobre");
    expect(contato).toHaveAttribute("href", "/contato");

    const nav = document.querySelector("nav");
    expect(nav).not.toBeNull();
    if (nav) {
      expect(nav.className).toMatch(/md:flex/);
    }
  });

  test("snapshot do Header", () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });
});
