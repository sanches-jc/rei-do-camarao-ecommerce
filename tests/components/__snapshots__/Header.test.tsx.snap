// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header component > snapshot do Header 1`] = `
<div>
  <header
    class="bg-white shadow-sm border-b border-gray-100"
  >
    <div
      class="container mx-auto px-4 py-4"
    >
      <div
        class="flex items-center justify-between"
      >
        <a
          class="flex items-center space-x-2"
          href="/"
        >
          <div
            class="text-2xl font-bold text-blue-600"
          >
            Rei do Camarão
          </div>
        </a>
        <nav
          class="hidden md:flex space-x-8"
        >
          <a
            class="text-gray-700 hover:text-blue-600 transition-colors duration-200 font-medium"
            href="/"
          >
            Home
          </a>
          <a
            class="text-gray-700 hover:text-blue-600 transition-colors duration-200 font-medium"
            href="/produtos"
          >
            Produtos
          </a>
          <a
            class="text-gray-700 hover:text-blue-600 transition-colors duration-200 font-medium"
            href="/sobre"
          >
            Sobre Nós
          </a>
          <a
            class="text-gray-700 hover:text-blue-600 transition-colors duration-200 font-medium"
            href="/contato"
          >
            Contato
          </a>
        </nav>
        <div
          class="flex items-center space-x-4"
        >
          <button
            aria-label="Abrir carrinho"
            class="p-2 text-gray-700 hover:text-blue-600 transition-colors"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="w-5 h-5"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 3h2l.4 2M7 13h10l4-8H5.4M7 13L5.4 5M7 13l-2.293 2.293c-.63.63-.184 1.707.707 1.707H17m0 0a2 2 0 100 4 2 2 0 000-4zm-8 2a2 2 0 11-4 0 2 2 0 014 0z"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </header>
</div>
`;
